iib/workers/tasks/build_merge_index_image.py

```diff
@@ -97,25 +102,22 @@ def _add_bundles_missing_in_source(
             and bundle['csvName'] not in source_bundle_csv_names
         ):
             missing_bundles.append(bundle)
+            missing_bundle_paths.append(bundle['bundlePath'])
 
-    missing_bundle_paths = [
-        bundle['bundlePath']
-        for bundle in itertools.chain(missing_bundles, source_index_bundles)
-        if is_bundle_version_valid(bundle['bundlePath'], ocp_version)
-    ]
+    for bundle in itertools.chain(missing_bundles, source_index_bundles):
+        if not is_bundle_version_valid(bundle['bundlePath'], ocp_version):
+            invalid_bundles.append(bundle)
 
-    if missing_bundle_paths:
-        log.info('%s bundles are missing in the source index image.', len(missing_bundle_paths))
-    else:
+    if invalid_bundles:
         log.info(
-            'No bundles are missing in the source index image. However, the index image is '
-            'still being rebuilt with the new binary image.'
+            '%s bundles have invalid version label and will be deprecated.', len(invalid_bundles)
```

Comment:
Will the actual invalid bundles be logged anywhere?

Reply:
Yes, when the `opm deprecatetruncate` command is called. It can be a large list and so I am trying to avoid logging it more than once